test/test_linalg.py

```diff
@@ -922,6 +924,256 @@ def test_nuclear_norm_exceptions_old(self, device):
         self.assertRaisesRegex(IndexError, "Dimension out of range", torch.norm, x, "nuc", (0, 2))
 
 
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @dtypesIfCPU(torch.float32, torch.float64, torch.complex64, torch.complex128)
+    @dtypesIfCUDA(torch.float32, torch.float64)
+    def test_cholesky(self, device, dtype):
+        from torch.testing._internal.common_utils import random_hermitian_pd_matrix
+
+        def run_test(shape, batch):
+            A = random_hermitian_pd_matrix(shape, *batch, dtype=dtype, device=device)
+            expected_L = np.linalg.cholesky(A.cpu().numpy())
+            actual_L = torch.linalg.cholesky(A)
+            matrices_are_equal = np.allclose(actual_L.cpu().numpy(), expected_L)
+
+            # For fp32 individual entries in matrices can differ between PyTorch and NumPy
```

Comment:
I'm worried about this approach. I think it's very clever but the discrepancy is concerning. 

Is this difference because NumPy is always computing the decomposition in double? Instead of this, should we compare in float with a lower precision, instead?

Reply:
Yes, NumPy is computing the result in double, then converts it to fp32.
https://github.com/numpy/numpy/blob/v1.19.0/numpy/linalg/linalg.py#L136

So the current approach is: try comparing individual entries with standard tolerances, if it fails then compare the matrices in norm and individual entries with a higher tolerance. Comparing matrices in norm gives us extra confidence that we get something approximately equal as matrix objects, even though the used tolerance (`atol=1e-2, rtol=1e-5`) for comparing individual values in the matrices may seem too high.

Reply:
OK. I'm convinced. That sounds good. Let's just switch NumPy's allclose for PyTorch's testcase's `assertEqual`.